### DISCUSSION TITLE
Running a web browser on the host with access to the user-v2 network?

### DISCUSSION THREAD

**Discussion by asgeirn** [lima-vm/lima#2705] [Q&A] [answered]
I'm running a Kubernetes cluster with one control plane and two worker nodes based on the oraclelinux-8 template and user-v2 networking.

This works beautifully with the exception of accessing services exposed as Ingress or LoadBalancer in the cluster - if I do not want do do SSH port forwarding.

Running Firefox on one of the VMs does work, but only intermittently as there seems to be some glitches in the X11 forwarding, and still isn't the preferred option.

Since the limactl usernet service is running on the host, my reasoning is that it should be possible to string some bits together to get an application to connect to this.

But where to get started?  The limactl usernet service does have an API, but I can't find any documentation on how to get started on this.

Any ideas?

**Comment by AkihiroSuda** [comment]
- Run `ssh -F ~/.lima/default/ssh.config -D 127.0.0.1:12345 lima-default`
- Open `System Settings → Network → Wi-Fi (or something else) → Details → Proxies → SOCKS proxy`, enter `Server: 127.0.0.1`, `Port: 12345`,
- Open `http://lima-<INSTANCE>.internal` via a Web browser

We should have a documentation for this.

**Reply by AkihiroSuda** [reply, marked as answer]
PR to add `limactl tunnel` command for wrapping the ssh command above:
- https://github.com/lima-vm/lima/pull/2710

**Reply by asgeirn** [reply]
Thanks, I missed that obvious solution :)

**Reply by fwilhe2** [reply]
I have a similar setup and I tried to reproduce the mentioned steps, but it does not seem to be working. Not sure what I did wrong. I'd be happy to try it out if there is a bit more documentation on the topic.

My other use-case would be to use kubectl on my host machine, like it's working in the single-node example for k8s. My understanding is that for a multi-node setup I need the v2-user networking, and this does not allow the host -> vm communication (without this tunnel), is that correct?

**Reply by AkihiroSuda** [reply]
You can still use the local host port forwarding as in the single-node k8s template
https://github.com/lima-vm/lima/blob/78ffc3ace0ae1da0ff16542c960b7acc1ba754b1/examples/k8s.yaml#L127

**Reply by AkihiroSuda** [reply]
On macOS you may use vmnet to assign “real” IP addresses that does not need  port forwarding nor proxying 
https://lima-vm.io/docs/config/network/#managed-192168105024

**Reply by asgeirn** [reply]
For `kubectl` and `k9s` on the host machine; `.kube/config` supports SOCKS5 proxying: 

```
clusters:
- cluster:
    server: https://192.168.104.3:6443
    proxy-url: socks5://localhost:12347
````
